### PR TITLE
schema changes for QRep BigQuery

### PR DESCRIPTION
**Pull request by heavycrystal**
Also fixes a logic bug where the loop for processing SchemaDeltas would end processing early if an empty SchemaDelta was found.

### CODE REVIEW

File: flow/connectors/bigquery/bigquery.go [hunk 1]
```diff
@@ -217,7 +217,7 @@ func (c *BigQueryConnector) ReplayTableSchemaDeltas(flowJobName string,
 	schemaDeltas []*protos.TableSchemaDelta) error {
 	for _, schemaDelta := range schemaDeltas {
 		if schemaDelta == nil || len(schemaDelta.AddedColumns) == 0 {
-			return nil
+			continue
```
Comment (serprex) [inline flow/connectors/bigquery/bigquery.go:220]: It'd be useful to explain why `return nil` becomes `continue` in PR description

File: flow/connectors/bigquery/bigquery.go [hunk 2]
```diff
@@ -1088,6 +1088,16 @@ func (c *BigQueryConnector) SetupNormalizedTables(
 ) (*protos.SetupNormalizedTableBatchOutput, error) {
 	tableExistsMapping := make(map[string]bool)
 	for tableIdentifier, tableSchema := range req.TableNameSchemaMapping {
+		table := c.client.Dataset(c.datasetID).Table(tableIdentifier)
```
Comment (serprex) [inline flow/connectors/bigquery/bigquery.go:1091]: Is this code movement only for optimization purposes & not correctness purposes?

File: flow/connectors/bigquery/bigquery.go (hunk 1 above)
Comment (heavycrystal) [reply flow/connectors/bigquery/bigquery.go:220]: added in the description, it is a bugfix

File: flow/connectors/bigquery/bigquery.go (hunk 2 above)
Comment (heavycrystal) [reply flow/connectors/bigquery/bigquery.go:1091]: only for optimization

**Review by serprex — APPROVED** [review]

**Review by iskakaushik — APPROVED** [review]

**Comment by iskakaushik** [comment]
@heavycrystal  why can there be an empty schema delta?

**Comment by heavycrystal** [comment]
currently there is no possibility for an empty SchemaDelta to be generated, but changed defensively

**Comment by heavycrystal** [comment]
closes #616

**Pull request merged**